test/test_selectors.py

```diff
@@ -0,0 +1,673 @@
+from __future__ import with_statement
+import errno
+import os
+import psutil
+import signal
+import socket
+import sys
+import time
+import threading
+
+try:  # Python 2.6 unittest module doesn't have skip decorators.
+    from unittest import skip, skipIf, skipUnless
+    import unittest
+except ImportError:
+    from unittest2 import skip, skipIf, skipUnless
+    import unittest2 as unittest
+
+try:  # Python 2.x doesn't define time.perf_counter.
+    from time import perf_counter as get_time
+except ImportError:
+    from time import time as get_time
+
+try:  # Python 2.6 doesn't have the resource module.
+    import resource
+except ImportError:
+    resource = None
+
+try:  # Windows doesn't support socketpair on Python 3.5<
+    from socket import socketpair
+except ImportError:
+    from .socketpair_helper import socketpair
+
+from urllib3.util import (
+    selectors,
+    wait_for_read,
+    wait_for_write
+)
+
+HAS_ALARM = hasattr(signal, "alarm")
+
+LONG_SELECT = 0.2
+SHORT_SELECT = 0.01
+
+# Tolerance values for CI timer/speed fluctuations.
```

Comment:
This comment should suggest that the tolerance is not for CI but for general timer/speed fluctuations.